src/ActionMenu.tsx

```diff
@@ -39,30 +39,21 @@ const ActionMenuBase = ({
   )
 
   const renderMenuItem: typeof Item = useCallback(
-    ({onClick, ...itemProps}) =>
-      renderItem({
+    ({onAction: itemOnAction, ...itemProps}) => {
+      const onActionWithClose = (
+        props: ItemProps,
+        event?: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>
+      ) => {
+        const actionCallback = itemOnAction ?? onAction
+        actionCallback?.(props as ItemProps, event)
+        onClose()
```

Comment:
So much cleaner!  This will need to be rebased in #1188, but it should be pretty easy on @T-Hugs side.